WordPressVIPMinimum/Sniffs/JS/WindowSniff.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * WordPressVIPMinimum_Sniffs_JS_WindowSniff.
+ *
+ * @package VIPCS\WordPressVIPMinimum
+ */
+
+namespace WordPressVIPMinimum\Sniffs\JS;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use PHP_CodeSniffer\Util\Tokens;
+
+/**
+ * WordPressVIPMinimum_Sniffs_JS_WindowSniff.
+ *
+ * Looks for instances of window properties that should be flagged.
+ *
+ * @package VIPCS\WordPressVIPMinimum
+ */
+class WindowSniff implements Sniff {
+
+	/**
+	 * A list of tokenizers this sniff supports.
+	 *
+	 * @var array
+	 */
+	public $supportedTokenizers = array(
+		'JS',
+	);
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		return [
+			T_STRING,
+		];
+	}
+
+	/**
+	 * List of window properties that need to be flagged.
+	 *
+	 * @var array
+	 */
+	private $windowProperties = [
+		'location' => [
+			'href'     => true,
+			'protocol' => true,
+			'host'     => true,
+			'hostname' => true,
+			'pathname' => true,
+			'protocol' => true,
+			'search'   => true,
+			'hash'     => true,
+			'username' => true,
+			'port'     => true,
+			'password' => true,
+		],
+		'name'   => true,
+		'status' => true,
+	];
+
+	/**
+	 * A list of tokens that are allowed in the syntax.
+	 *
+	 * @var array
+	 */
+	private $syntaxTokens = [
+		T_OBJECT_OPERATOR,
+
+	];
+
+	/**
+	 * Processes this test, when one of its tokens is encountered.
```

Comment:
I feel like this comment needs some flair explaining what it's goal is; for example…

```
                                                                                                                           
                                                                                                                           
                                                                        ___             ___                                
                                                                        `MM             `MM                                
                  /                                /                     MM              MM                                
  __      ____   /M             _____  ___   ___  /M           ___   ___ MM  __          MM  __     ____  ___  __   ____   
 6MMbMMM 6MMMMb /MMMMM         6MMMMMb `MM    MM /MMMMM        `MM    MM MM 6MMb         MM 6MMb   6MMMMb `MM 6MM  6MMMMb  
6M'`Mb  6M'  `Mb MM           6M'   `Mb MM    MM  MM            MM    MM MMM9 `Mb        MMM9 `Mb 6M'  `Mb MM69 " 6M'  `Mb 
MM  MM  MM    MM MM           MM     MM MM    MM  MM            MM    MM MM'   MM        MM'   MM MM    MM MM'    MM    MM 
YM.,M9  MMMMMMMM MM           MM     MM MM    MM  MM            MM    MM MM    MM        MM    MM MMMMMMMM MM     MMMMMMMM 
 YMM9   MM       MM           MM     MM MM    MM  MM            MM    MM MM    MM        MM    MM MM       MM     MM       
(M      YM    d9 YM.  ,       YM.   ,M9 YM.   MM  YM.  ,        YM.   MM MM    MM        MM    MM YM    d9 MM     YM    d9 
 YMMMMb. YMMMM9   YMMM9        YMMMMM9   YMMM9MM_  YMMM9         YMMM9MM_MM_  _MM_      _MM_  _MM_ YMMMM9 _MM_     YMMMM9  
6M    Yb                                                                                                                   
YM.   d9                                                                                                                   
 YMMMM9                                                                                                                    
```